### ISSUE TITLE
Migrate off camlp4

### ISSUE THREAD

**Issue by XVilka** [mor1/ocaml-tftp#6]
Camlp4 is deprecated for a long time, Camlp5 as well. It makes sense either migrate to mehrir, or extension points for preprocessing and parsing capabilities.

See discussion about Camlp4, Camlp5, Ulex and modern OCaml here: https://discuss.ocaml.org/t/camlp5-and-ocaml-4-08/3985

Also it makes sense to switch to [Dune](https://dune.build) build system as well, plus adding a [CI scripts](https://github.com/ocaml/ocaml-ci-scripts).

**Comment by mor1** [comment]
Agree; not been active on this repo for a while. Might manage to pick it up again in the next few months. In the meantime, PRs welcome :)
